caas/kubernetes/provider/builtin.go

```diff
@@ -69,24 +72,18 @@ func getLocalMicroK8sConfig(cmdRunner CommandRunner) ([]byte, error) {
 	if err != nil {
 		return []byte{}, errors.NotFoundf("microk8s")
 	}
-	execParams := exec.RunParams{
-		Commands: "microk8s config",
+	notSupportErr := errors.NewNotSupported(nil, fmt.Sprintf("juju %q can only work with strict confined microk8s", version.Current))
+	snapDataPath := os.Getenv("SNAP_DATA")
+	if snapDataPath == "" {
+		return nil, errors.Annotate(notSupportErr, "SNAP_DATA is empty")
 	}
-	result, err := cmdRunner.RunCommands(execParams)
-	if err != nil {
-		return []byte{}, err
+	clientConfigPath := filepath.Join(snapDataPath, "credentials", "client.config")
```

Comment:
SNAP_DATA is a generic path, we need to include "microk8s" in the final clientConfigPath so that other paths can also be added to SNAP_DATA later